### PR TITLE
Users could find their location at the beginning map.

### PR DESCRIPTION
**Pull request by willis50317**
This pull request makes the following changes:
-
This branch add a feature to allow users could find their location by pressing the location button in the beginning map.
Basically, we only revised the file "map.js" (/platform-client/app/common/services/map.js).

picture_uplus_client_locate

Test these changes by:
-
OS: Ubuntu 16.04 
Browser: Firefox, Chrome
Use vagrant to build a local website to test this feature.

Fixes ushahidi/platform# .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/426)

### CODE REVIEW

**Comment by willis50317** [comment]
[clabot:check]

***********************************************************************
*Li-Wei, Tsai*
*Master* in *Engineering Science*
System Control Lab, Department of Engineering Science
National Cheng Kung University, Taiwan
Research Field: Dynamics, System Control, System Identification
E-mail: willis50317@gmail.com <william60505@gmail.com>
***********************************************************************

2016-12-01 16:33 GMT+08:00 Ushbot <notifications@github.com>:

**Comment by willis50317** [comment]
Dear Ushahidi,
Sorry as a beginner, I would like to ask a question.
Will you inform me once you check whether this branch could be merged?
Liwei

**Comment by tuxpiper** [comment]
Dear @willis50317 , thank you very much for your contribution. Once we review and approve this pull request, you would get a notification.

However, note that the develop branch has changed since your submission and now there seem to be some merge conflicts. That may delay things a bit.

**Comment by willis50317** [comment]
@tuxpiper Hello David! Thanks for your reply!
I would like to help anything I could to accelerate this process. 
I also noticed these conflicts are wired. I only added codes in "map.js". I expect it can be merged. 
Could you provide me a way to update my platform-client package to the latest official version?
Maybe It could be merged while I make changes on the latest version.

**Comment by willis50317** [comment]
@tuxpiper 
that's wired... I changed "scrollWheelZoom" to be false as its original value. 
Still doesn't work...

**Comment by willis50317** [comment]
@tuxpiper 
Hello could you help me to fix the failure of Travis CI?
It looks like there is an unexpected token at line 108 of "map.js". 
However, I didn't make any change in that line and keep it as origin.

**Comment by rjmackay** [comment]
@willis50317 sorry we've been slow on this. I'll try to review next week. The likely issue here is the whether this is actually a behaviour change we want in core, I'll check it out and see what @jshorland thinks

File: app/common/services/maps.js [hunk 1]
```diff
@@ -103,6 +103,76 @@ function Maps(ConfigEndpoint, L, _, CONST) {
             if (config.default_view.baselayer === 'MapQuest') {
                 config.default_view.baselayer = 'streets';
             }
+            return deferred.promise;
+        },
+        reloadMapConfig: function () {
```
Comment (rjmackay) [inline app/common/services/maps.js]: It looks like this has brought in some old code? this function isn't used or needed

File: app/common/services/maps.js [hunk 2]
```diff
@@ -103,6 +103,76 @@ function Maps(ConfigEndpoint, L, _, CONST) {
...
+                if (config.default_view.baselayer === 'MapQuestAerial') {
+                    config.default_view.baselayer = 'satellite';
+                }
+                this.config = config;
+                return this.config;
+            }, this));
+        }
+    };
+
+    var Map = {
```
Comment (rjmackay) [inline app/common/services/maps.js]: again. It looks like most of this is out of date. In the current code, I think we could just add this to `createMap` and move the code from posts/modify/location.directive.js thus adding the locate control to *all* our maps

**Pull request closed without merging**